sycl/source/detail/graph_impl.cpp

```diff
@@ -913,6 +912,149 @@ exec_graph_impl::enqueue(const std::shared_ptr<sycl::detail::queue_impl> &Queue,
   return QueueEvent;
 }
 
+void exec_graph_impl::duplicateNodes() {
+  // Map of original modifiable nodes (keys) to new duplicated nodes (values)
+  std::map<std::shared_ptr<node_impl>, std::shared_ptr<node_impl>> NodesMap;
+
+  std::vector<std::shared_ptr<node_impl>> ModifiableNodes =
+      MGraphImpl->MNodeStorage;
+  std::deque<std::shared_ptr<node_impl>> NewNodes;
+
+  for (int i = ModifiableNodes.size() - 1; i >= 0; i--) {
```

Comment:
Since we are now linking nodes to their successors in another loop, I think we do not need to loop back for creating nodes. Using `for (int i = 0; i < ModifiableNodes.size();  i++) {` might make the code easier to understand.

Reply:
Good point! Fixed in both locations.